docs/source/reference/api-server/api-server-upgrade.rst

```diff
@@ -0,0 +1,204 @@
+
+1. Patch the API server to remove it from the Ingress backend:
+
+.. note::
+
+    All new requests will be rejected by the Ingress after this step. Make sure there is no critical service depending on the API server before proceeding.
+
+.. code-block:: bash
+
+    kubectl get pod -l app=${RELEASE_NAME}-api -oname | xargs kubectl patch --type merge -p '{"metadata": {"labels": {"skypilot.co/ready": null}}}'
+
+2. Verify the API server is cordoned, you should see the following error:
+
+.. code-block:: console
+
+    $ sky api info
+    sky.exceptions.ApiServerConnectionError: Could not connect to SkyPilot API server at <URL>. Please ensure that the server is running. Try: curl <URL>
+
+.. dropdown:: Resolve cordon failure for early 0.8.0 nightly release
+
+    If you are upgrading from early 0.8.0 nightly release that does not support cordoning (``sky api info`` will succeed), you can manually enable cordon support by running:
+
+    .. code-block:: bash
+
+        kubectl patch service ${RELEASE_NAME}-api-service -p '{"spec":{"selector":{"skypilot.co/ready":"true"}}}'
+    
+    After the patch, verify the API server is cordoned again.
+
+3. For requests that have been sent to the API server, check the status of the requests and wait for the critical ones to finish:
+
+.. code-block:: console
+
+    $ kubectl get po -l app=${RELEASE_NAME}-api -oname | xargs -I {} kubectl exec {} -c skypilot-api -- sky api status
+    sky api status
+    ID                                    User             Name        Created         Status
+    942f6ab3-f5b6-4a50-acd6-0e8ad64a3ec2  <USER>           sky.launch  a few secs ago  PENDING
+    8c5f19ca-513c-4068-b9c9-d4b7728f46fb  <USER>           sky.logs    26 secs ago     RUNNING
+    skypilot-status-refresh-daemon        skypilot-system  sky.status  25 mins ago     RUNNING
+
+The `skypilot-status-refresh-daemon` is a background process managed by API server that can be safely interrupted. For other requests, you can either wait for a request to finish or cancel a request by running:
+
+.. code-block:: console
+
+    $ kubectl get po -l app=${RELEASE_NAME}-api -oname | xargs -I {} kubectl exec {} -c skypilot-api -- sky api cancel ${ID}
+
+If you do not have `exec` access to the API server Pod, you can also use `port-forward` to access the api status:
+
+.. code-block:: console
+
+    $ kubectl get po -l app=${RELEASE_NAME}-api -oname | xargs -I {} kubectl port-forward {} 46580:46580 > /tmp/port-forward.log 2>&1 &
+    $ PORT_FORWARD_PID=$!
+    $ sky api login -e http://127.0.0.1:46580
+    # Polling the status
+    $ sky api status
+    # Cancel less critical requests if needed
+    $ sky api cancel ${ID}
+    # Stop the port-forward after you are satisfied with the status
+    $ kill $PORT_FORWARD_PID
```

Comment:
How about we just have it in a dropdown? It does not seem to be a very critical thing to mention.

Reply:
Sounds good, will do